multiplex.go

```diff
@@ -34,13 +34,30 @@ var ErrTwoInitiators = errors.New("two initiators")
 // In this case, we close the connection to be safe.
 var ErrInvalidState = errors.New("received an unexpected message from the peer")
 
+var errTimeout = timeout{}
+var errStreamClosed = errors.New("stream closed")
+
 var (
 	NewStreamTimeout   = time.Minute
 	ResetStreamTimeout = 2 * time.Minute
 
 	WriteCoalesceDelay = 100 * time.Microsecond
 )
 
+type timeout struct{}
+
+func (_ timeout) Error() string {
+	return "timeout"
```

Comment:
"i/o deadline exceeded"? (or one for read/write?)

Reply:
yeah, that's better. will do.

Reply:
done